Makefile

```
@@ -159,6 +162,18 @@ CREATE_KIND_CLUSTER ?= true
 test-e2e: kustomize manifests generate envtest ginkgo
 	E2E_KIND_VERSION=$(E2E_KIND_VERSION) KIND_CLUSTER_NAME=$(KIND_CLUSTER_NAME) CREATE_KIND_CLUSTER=$(CREATE_KIND_CLUSTER) ARTIFACTS=$(ARTIFACTS) IMAGE_TAG=$(IMAGE_TAG) ./hack/e2e-test.sh
 
+E2E_TARGETS := $(addprefix run-test-e2e-,${E2E_K8S_VERSIONS})
+.PHONY: test-e2e-all
+test-e2e-all: kustomize manifests generate envtest ginkgo cleanup-e2e $(E2E_TARGETS)
```

Could we do something like this?

```suggestion
test-e2e-all: kustomize manifests generate envtest ginkgo
    for E2E_KIND_VERSION in $E2E_TARGETS; do \
        run-test-e2e; \
    done
```

---

I would suggest using `foreach` like the following:

```suggestion
E2E_K8S_VERSIONS := v1.24 v1.25 v1.26
.PHONY: test-e2e-all
test-e2e-all: kustomize manifests generate envtest ginkgo
	$(foreach VERSION,$(E2E_K8S_VERSIONS),$(MAKE) run-test-e2e-$(VERSION);)
```

---

The `foreach` is a built-in function in make.

---

I don't think I see any benefits coming out of it.

---

I was trying to get rid of `cleanup-e2e`. It looks quite hacky that we have to delete some files.

---

> I was trying to get rid of cleanup-e2e. It looks quite hacky that we have to delete some files.

+1

We should look into a way to avoid adding `cleanup-e2e`.

---

`cleanup-e2e` avoided